### PR TITLE
feature: 制品分析的扫描报告和制品漏洞报告导出 #830

### PR DESCRIPTION
**Pull request by SunnyShen434**
What would you like to be added:
1. 扫描任务状态添加UN_QUALITY、QUALITY_PASS、QUALITY_UNPASS, 以便于扫描报告导出; 并同步其他扫描任务状态的处理逻辑
2. 制品分析的扫描报告导出
3. 制品分析的制品漏洞报告导出
4. 修复函数命名错误: createProjectIdAdnRepoRule -> createProjectIdAndRepoRule
5. 之前代码提交不完整, 补充修复: 删除仓库后查看制品扫描记录鉴权异常

Why is this needed:
作为制品库管理员, 需要查看安全扫描报告, 以便于做数据统计与汇报

### CODE REVIEW

**Comment by tencentblueking-adm** [comment]
[CLA assistant check](https://bk-cla.bktencent.com:443/TencentBlueKing/bk-repo?pullRequest=831) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://bk-cla.bktencent.com:443/TencentBlueKing/bk-repo?pullRequest=831) before we can accept your contribution.<br/><hr/>**xiaoyushen** seems not to be a GitHub user. You need a GitHub account to be able to sign the CLA. If you have already a GitHub account, please [add the email address used for this commit to your account](https://help.github.com/articles/why-are-my-commits-linked-to-the-wrong-user/#commits-are-not-linked-to-any-user).<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://bk-cla.bktencent.com:443/check/TencentBlueKing/bk-repo?pullRequest=831) it.</sub>

**Review by cnlkl — CHANGES_REQUESTED** [review]

File: src/backend/analyst/biz-analyst/src/main/kotlin/com/tencent/bkrepo/analyst/controller/user/UserScanReportController.kt [hunk 1]
```diff
@@ -109,6 +109,19 @@ class UserScanReportController(private val scanTaskService: ScanTaskService) {
         return ResponseBuilder.success(scanTaskService.resultDetail(request))
     }
 
+    @ApiOperation("制品详情-漏洞数据-导出")
+    @GetMapping("/export/artifact/leak/{projectId}/{subScanTaskId}")
```
[the remark this answers is not in the document]
Comment (SunnyShen434) [reply src/backend/analyst/biz-analyst/src/main/kotlin/com/tencent/bkrepo/analyst/controller/user/UserScanReportController.kt:113]: 此接口与`/artifact/leak/{projectId}/{subScanTaskId}`接口类似, 都调用`fun resultDetail(request: ArtifactVulnerabilityRequest): Page<ArtifactVulnerabilityInfo>`函数, 那里面已做鉴权, 不再重复鉴权

File: src/backend/analyst/biz-analyst/src/main/kotlin/com/tencent/bkrepo/analyst/service/impl/ScanTaskServiceImpl.kt [hunk 2]
```diff
@@ -141,6 +146,41 @@ class ScanTaskServiceImpl(
...
+        logger.info("exportScanPlanRecords request:${request.toJsonString()}")
+        with(request) {
+            if (id == null) {
+                throw ErrorCodeException(CommonMessageCode.PARAMETER_INVALID)
+            }
+            // 方案信息，获取方案名称
+            val scanPlan = scanPlanDao.find(projectId, id!!)
+                ?: throw ErrorCodeException(CommonMessageCode.PARAMETER_INVALID)
+
+            val isLicenseScan = scanPlan.type.endsWith("_LICENSE")
```
[the remark this answers is not in the document]
Comment (SunnyShen434) [reply src/backend/analyst/biz-analyst/src/main/kotlin/com/tencent/bkrepo/analyst/service/impl/ScanTaskServiceImpl.kt:159]: 已修改, 感谢指正

File: src/backend/analyst/biz-analyst/src/main/kotlin/com/tencent/bkrepo/analyst/utils/ScanLicenseConverter.kt [hunk 3]
```diff
@@ -43,6 +44,26 @@ import java.time.format.DateTimeFormatter
...
+    // 报告许可总数
+    private const val LICENSE_TOTAL = "total"
+
+    fun convert(subScanTask: TPlanArtifactLatestSubScanTask): LicenseScanPlanExport {
+        with(subScanTask) {
+            return LicenseScanPlanExport(
+                name = artifactName,
+                versionOrFullPath = version ?: fullPath,
+                repoName = repoName,
+                qualityRedLine = qualityRedLine?.let { if (it) "通过" else "不通过" } ?: "/",
```
Comment (cnlkl) [inline src/backend/analyst/biz-analyst/src/main/kotlin/com/tencent/bkrepo/analyst/utils/ScanLicenseConverter.kt:56]: 可以考虑用`LocaleMessageUtils`获取`通过` `不通过`的文案

**Review by cnlkl — CHANGES_REQUESTED** [review]

**Comment by cnlkl** [comment]
#471

File: src/backend/analyst/biz-analyst/src/main/kotlin/com/tencent/bkrepo/analyst/service/impl/ScanTaskServiceImpl.kt [hunk 4]
```diff
@@ -148,37 +150,31 @@ class ScanTaskServiceImpl(
...
+        var page = planArtifactLatestSubScanTaskDao.pageBy(request)
+        val dataList = mutableListOf<Any>()
+        while (page.records.isNotEmpty()) {
+            page.records.forEach {
+                dataList.add(
+                    if (isLicenseScan) ScanLicenseConverter.convert(it)
+                    else Converter.convertToPlanExport(it)
+                )
             }
+            request.pageNumber = ++pageNumber
```
[the remark this answers is not in the document]
Comment (SunnyShen434) [reply src/backend/analyst/biz-analyst/src/main/kotlin/com/tencent/bkrepo/analyst/service/impl/ScanTaskServiceImpl.kt:172]: 已在`planArtifactLatestSubScanTaskDao`中添加一个`获取指定扫描方案的制品最新扫描记录`的方法, 感谢指正

File: src/backend/analyst/biz-analyst/src/main/kotlin/com/tencent/bkrepo/analyst/utils/ScanLicenseConverter.kt (hunk 3 above)
Comment (SunnyShen434) [reply src/backend/analyst/biz-analyst/src/main/kotlin/com/tencent/bkrepo/analyst/utils/ScanLicenseConverter.kt:56]: 考虑到前后端一致性, 需要删除`qualityRedLine`字段, 使用`status`导出为扫描状态`ScanStatus`;
已修改为用`LocaleMessageUtils`获取8种扫描状态的文案. 感谢指正

File: src/backend/analyst/biz-analyst/src/main/kotlin/com/tencent/bkrepo/analyst/service/impl/ScanTaskServiceImpl.kt [hunk 5]
```diff
@@ -302,7 +374,12 @@ class ScanTaskServiceImpl(
     ): FileLicensesResultOverview {
         val subtask = subtaskDao.findById(subScanTaskId)
             ?: throw NotFoundException(CommonMessageCode.RESOURCE_NOT_FOUND, subScanTaskId)
-        permissionCheckHandler.checkSubtaskPermission(subtask, PermissionAction.READ)
+        try {
+            permissionCheckHandler.checkSubtaskPermission(subtask, PermissionAction.READ)
+        } catch (e: UncheckedExecutionException) {
```
Comment (cnlkl) [inline src/backend/analyst/biz-analyst/src/main/kotlin/com/tencent/bkrepo/analyst/service/impl/ScanTaskServiceImpl.kt:379]: 这里什么情况会抛出`UncheckedExecutionException`

Comment (SunnyShen434) [reply src/backend/analyst/biz-analyst/src/main/kotlin/com/tencent/bkrepo/analyst/service/impl/ScanTaskServiceImpl.kt]: 获取许可扫描子任务信息 的时候会抛`UncheckedExecutionException`, 但是现在版本已经没有在使用这个函数相关的接口, 所以代码无需改动
已移除, 感谢指正!

**Review by cnlkl — APPROVED** [review]

**Review by cnlkl — APPROVED** [review]

**Review by cnlkl — APPROVED** [review]

**Pull request merged**